### DISCUSSION TITLE
can't use with google gemini provided throught the .env file

### DISCUSSION THREAD

**Discussion by delta-whiplash** [danny-avila/LibreChat#1828] [Troubleshooting] [answered]
### What happened?

I added my apikey in my .env file via the GOOGLE_KEY variable but when I try to generate something I have this error : 
Something went wrong. Here's the specific error message we encountered: Cannot read properties of undefined (reading 'apiKey')

It's not a network localisation error because I can generate text from the container with curl  and my api key

if I set the GOOGLE_KEY variable to user_provided and I provide it through the web GUI I can generate without any issue

By the way I tryed a setup with the variable with my apikey in .env  yesterday and it worked, am using the latest docker image from git

### Steps to Reproduce

1. setup the api key in the .env file 
2. deploy the container
3. try to generate something with the webgui using gemini-pro model

### What browsers are you seeing the problem on?

Chrome

### Relevant log output

```shell
2024-02-17 15:26:11 error: [handleAbortError] AI response error; aborting request: Cannot read properties of undefined (reading 'apiKey')
```

### Screenshots

![image](https://github.com/danny-avila/LibreChat/assets/39416418/ab055876-be32-4e81-9376-9af690da19bd)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
working on my end (both local with npm and docker): 
![image](https://github.com/danny-avila/LibreChat/assets/110412045/0a18783a-fc8f-4d39-83ac-0d7c669b1301)

Where are you deploying the container?

**Reply by delta-whiplash** [reply]
I deployed it on kubernetes, my .env and librechat.yaml are provided through configmap in read_only Access mode 

how can I debug to find the origin of my issue ?

**Reply by that-scientist** [reply]
I wonder if google flipped over which API key they want?  I may be mis-remembering this, but I thought I initially set up with a generic project API key that stopped working.  I just set up with a Google AI Studio API key and it's working as expected now.

**Comment by danny-avila** [comment]
If your other env file values are being read, it should be working. Post you .env with sensitive values redacted as is

**Reply by delta-whiplash** [reply]
I tryed this afternoon and it's working again without touching anything

**Reply by danny-avila** [reply]
Sweet unsure why it wouldn’t have worked for you

**Reply by fuegovic** [reply]
Today I got the same error `Cannot read properties of undefined (reading 'apiKey')` on multiple endpoints, (OpenAI, Assistants, Plugins, Google...) 

I have no idea what's the trigger, sometimes it works fine, sometimes I get the error 🤷‍♂️

note: I tested with multiple API keys on 2 accounts, everything is set as `user_provided`

**Comment by that-scientist** [comment]
This might not be specific to Gemini.  I've been getting similar errors on the OpenAI Assistants API today.  After changing the key it worked for a few short prompts, then started throwing an error after I tried to process a large chunk of text pasted into the chat interface.
 
Running LibreChat via the docker setup on a DO droplet.  

OpenAI ChatGPT works fine, but assistant interface gives this error:
"Something went wrong. Here's the specific error message we encountered: API key not provided."

**Reply by danny-avila** [reply]
I’ll investigate and see if I can recreate it

**Reply by that-scientist** [reply]
Thanks!  If you'd like a login to my instance, just message me directly (joined your discord as well, where I'm doc / psychdev).  As of now, same error message there when trying the assistants API.  Openrouter, Google, and OpenAI are working fine.

**Comment by danny-avila** [comment, marked as answer]
Merged a fix for the primary issue here: https://github.com/danny-avila/LibreChat/pull/1834
